File: esykart-apigateway/src/main/java/com/webientsoft/esykart/apigateway/service/RoleService.java
package com.webientsoft.esykart.apigateway.service;

import com.webientsoft.esykart.common.model.Status;
import com.webientsoft.esykart.common.model.common.FilterModel;
import com.webientsoft.esykart.common.model.common.PaginatedDataModel;
import com.webientsoft.esykart.common.model.user.RoleModel;

public interface RoleService {

	void save(RoleModel model);

	void update(Integer id, RoleModel model);

	RoleModel find(Integer id);

	PaginatedDataModel search(FilterModel model);

	void changeStatus(Integer id, Status status);

}

File: esykart-store/esykart-store-merchant/src/main/resources/public/app/components/role/js/roleService.js
angular.module('altairApp').constant('roleServiceUrl', 'roles')
.service('RoleService',
['$http','roleServiceUrl',
function($http,roleServiceUrl) {
	var service = {};
	service.save = function (data){
    	return $http.post(roleServiceUrl,role);
    };
    service.update = function(id,data){
		return $http.put(roleServiceUrl+"/" +roleId ,role);
    };
    
    service.find = function(id,projection){
		return $http.get(roleServiceUrl +"/" +id+"?projection="+projection);
    };
    service.findAll = function(page, size,sort,projection){
    	var finalUrl = roleServiceUrl+"?projection="+projection;
		finalUrl += "&page="+page;
		finalUrl += "&size="+size;
	 	if(sort!= null && sort.length > 0){
			$(sort).each(function(){
				finalUrl += "&sort=";
				finalUrl += this;
			});
		}
		return $http.get(finalUrl);
	};

    service.search = function(filter,projection){
		return $http.get(roleServiceUrl +"/search?projection="+projection,filter);
    };
    
    service.changeStatus = function(id,status){
    	return $http.patch(roleServiceUrl +"/" + id + "/changeStatus?status="+status);
    }
    return service;
} 
]);
